### ISSUE TITLE
Rails Cache and `expire_in` support

### ISSUE THREAD

**Issue by sandstrom** [rails/rails#34015]
### Problem

According to the docs `:expire_in` can be used with all cache stores (see below), but it doesn't actually work. For example, it won't work with the file store (possibly not with the memory store either).

> This class [ActiveSupport::Cache::Store] provides the foundation for interacting with the cache in Rails. This is an abstract class and you cannot use it on its own. Rather you must use a concrete implementation of the class tied to a storage engine. Rails ships with several implementations documented below. […]
>
> There are some common options used by all cache implementations. These can be passed to the constructor or the various methods to interact with entries. […]
> 
> - `:expires_in` - This option sets an expiration time in seconds for the cache entry when it will be automatically removed from the cache.
> 
> SOURCE: https://guides.rubyonrails.org/caching_with_rails.html#cache-stores

### Steps to reproduce

```ruby
# setup in config
config.cache_store = :file_store

# verify current store (console)
Rails.cache.class #=> ActiveSupport::Cache::FileStore

# read-or-set a value, wait 3 seconds, read again (and get the old timestamp)
Rails.cache.fetch('foo', :expire_in => 2.seconds) { Time.now } #=> some timestamp
sleep 3
Rails.cache.fetch('foo', :expire_in => 2.seconds) { Time.now } #=> same as above (not expected)
```

### Possible solutions
I understand if some cache stores may not want to evict data. In these cases one could instead store the expire timestamp along with the data, so when it's read a second time (after expiry) we'll replace the entry.

Basically, I think it makes sense to separate eviction from expiry. It's fine if all stores don't evict, but all of them should respect `:expire_in` for reading.

Otherwise, use-cases such as refreshing data from an external service cannot use the cache.

```ruby
# Instead of tweets this could also be exchange rates, football scores, system status, 
# maintenance state of sub-components, and a lot of other things.
Rails.cache.fetch('recent_tweets', :expire_in => 1.minute) do
  fetch_recent_tweets(user_id)
end
```

### System configuration
**Rails version**: 4.2
**Ruby version**: 2.5

**Comment by rafaelfranca** [comment]
As an issue I think we should update the documentation to `"There are some common options that can be used by all cache implementations."`

About the solution you proposed I'm fine with supporting something like this, but this is more a feature request and we don't take feature requests in the issue tracker.

**Comment by sandstrom** [comment]
@rafaelfranca Thanks, I saw the updated docs, awesome!

Regarding a change, is there any other forum besides the mailing list for proposing features/changes? 
(RFC-repo? IRC/Discord/Slack channel?)
